lectures/10.HPC_and_parallel_compute/scripts/analyze_sequences.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+import pathlib
+
+import pandas as pd
+
+
+def analyze_sequences(sequence: str) -> tuple:
+    """
+    This function analyzes a sequence for CpG content and returns the results.
+
+    Parameters
+    ----------
+    sequence : string
+        a string of DNA sequence
+
+    Returns
+    -------
+    c_count : int
+        the number of CpG sites in the sequence
+    raw_count : int
+        the total number of C sites in the sequence
+    """
+
+    # cast the sequence to uppercase
+    sequence = sequence.upper()
+    raw_count = 0
+    c_count = 0
+    cpg_count = 0
+    for i in enumerate(sequence):
+        if i[1] == "X":
```

Comment:
```suggestion
    for sequence_element in enumerate(sequence):
        if sequence_element[1] == "X":
```

Reply:
It's unclear to me how this is actually counting anything You're not updating cpg_count? What is "X"?